### PR TITLE
Basic upload file and get all siafiles test

### PR DESCRIPTION
**Pull request by agundy**
Creating pull request to show progress and get feedback. Especially looking for a recommended way of storing the json for testing.

### CODE REVIEW

**Comment by triazo** [comment]
Noticed that you're using an actual siafile to test. It seems like a good way of doing things for now. If you want to modify them, maybe make one shorter while still being valid, or parse them, remember that it's just gzipped json. You could probably cut out many of the redundant servers from inside the siafile to make it shorter. Command line utilities gzip and gunzip are good for this. If you wanted to parse this inside sunfish you can use http://golang.org/pkg/compress/gzip/ to do it on the fly.

File: sunfish_test.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+)
+
+// TestGetSiafiles test whether the api endpoint returns an empty list of
+// siafiles to start
+func TestGetSiafiles(t *testing.T) {
+	var siafiles []Siafile
+	req, err := http.NewRequest("GET", url+"/siafile/", nil)
+
+	if err != nil {
+		sf.logger.Println(err)
```
Comment (DavidVorick) [inline sunfish_test.go:28]: t.Fatal(err)

**Comment by DavidVorick** [comment]
In general it looks pretty good

**Comment by DavidVorick** [comment]
bump

**Comment by agundy** [comment]
Switch the logging line to a fail. Sorry California has kept me busy

**Pull request merged**